core/src/main/java/hudson/model/AbstractCIBase.java

```diff
@@ -89,17 +89,26 @@ protected void killComputer(Computer c) {
         c.kill();
     }
 
-    /* =================================================================================================================
-    * Package-protected, but accessed API
-    * ============================================================================================================== */
-
-    /*package*/ final CopyOnWriteArraySet<String> disabledAdministrativeMonitors = new CopyOnWriteArraySet<>();
+    private volatile Set<String> disabledAdministrativeMonitors = new HashSet<>();
```

Comment:
My mistake; this will likely still need to be either `CopyOnWriteArraySet` or use `Collections.synchronizedSet()` around this. Otherwise, a caller that iterates on the result of `getDisabledAdministrativeMonitors()` may get a `ConcurrentModificationException` when another thread calls `disable()` which modifies the set.

Alternatively, you can update `getDisabledAdministrativeMonitors` to wrap the set in a `new HashSet` to copy the set and prevent the issue entirely.

Reply:
> Alternatively, you can update getDisabledAdministrativeMonitors to wrap the set in a new HashSet to copy the set and prevent the issue entirely.

That doesn't work as the set is modified through the getter, by calling add on it.